src/index.js

```diff
@@ -658,20 +658,37 @@ export class DownloaderHelper extends EventEmitter {
     __getFileNameFromHeaders(headers, response) {
         let fileName = '';
 
+        const fileNameAndEncodingRegExp = /.*filename\*=.*?''([^"].+?[^"])(?:(?:;)|$)/i // match everything after the specified encoding behind a case-insensitive `filename*=`
+        const fileNameWithQuotesRegExp = /.*filename="(.*?)";?/i // match everything inside the quotes behind a case-insensitive `filename=`
+        const fileNameWithoutQuotesRegExp = /.*filename=([^"].+?[^"])(?:(?:;)|$)/i // match everything immediately after `filename=` that isn't surrounded by quotes and is followed by either a `;` or the end of the string
+        
+        const ContentDispositionHeaderExists = headers.hasOwnProperty('content-disposition')
+        const fileNameAndEncodingMatch = !ContentDispositionHeaderExists ? null : headers['content-disposition'].match(fileNameAndEncodingRegExp)
```

Comment:
One small change, if I'm understanding [this RFC](https://datatracker.ietf.org/doc/html/rfc5987#section-3.2.2) document correctly, the `filename*=` encoding property can have another element that isn't accounted for in the matcher, like so:  
`filename*=iso-8859-1'en'%A3%20rates`

The key part being the `en`. I think we need this matcher instead:
```regex
/.*filename\*=.*?'.*?'([^"].+?[^"])(?:(?:;)|$)/i
```

Which adds `.*?` in between the single quotes.

Reply:
Ah, I was wondering about those quotes, looked like a strange syntax to me!  
Good catch, I'll change it later :)